hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -278,6 +279,32 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, fmt.Errorf("failed to get cluster %q: %w", req.NamespacedName, err)
 	}
 
+	res, err := r.reconcile(ctx, req, log, hcluster)
+	condition := metav1.Condition{
+		Type:               string(hyperv1.ReconciliationSucceeded),
+		ObservedGeneration: hcluster.Generation,
+		Status:             metav1.ConditionTrue,
+		Message:            "Reconciliaton Succeeded",
+	}
+	if err != nil {
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = "ReconciliationError"
+		condition.Message = err.Error()
+	}
+	old := meta.FindStatusCondition(hcluster.Status.Conditions, string(condition.Type))
+	if old != nil {
+		old.LastTransitionTime = condition.LastTransitionTime
+	}
+	if !reflect.DeepEqual(old, condition) {
+		if statusUpdateErr := r.Client.Status().Update(ctx, hcluster); statusUpdateErr != nil {
+			return res, utilerrors.NewAggregate([]error{err, statusUpdateErr})
```

Comment:
couldn't you drop the if statusUpdateErr and just 
`return  res, r.Client.Status().Update(ctx, hcluster);` here?

Reply:
Good idea, fixed